src/firstboot/firstboot.c

```diff
@@ -349,12 +369,11 @@ static int process_locale(void) {
 
         locales[i] = NULL;
 
-        (void) mkdir_parents(etc_localeconf, 0755);
-        r = write_env_file(etc_localeconf, locales);
+        r = write_env_file_at(pfd, f, locales);
         if (r < 0)
-                return log_error_errno(r, "Failed to write %s: %m", etc_localeconf);
+                return log_error_errno(r, "Failed to write /etc/locale.conf: %m");
```

Comment:
Hmm, so previously the message showed the real path. That is very useful, when looking at a message. Please insert the chroot path in the message.